src/kbc_modules/cc_kbc/mod.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::kbc_modules::{KbcCheckInfo, KbcInterface};
+
+mod attester;
+mod crypto;
+mod kbs_protocol;
+
+use anyhow::*;
+use attester::{detect_tee_type, Attester, TEE};
+use crypto::{hash_chunks, TeeKey};
+use kbs_protocol::message::*;
+
+pub struct Kbc {
+    tee: String,
+    token: Option<String>,
+    nonce: String,
+    tee_key: Option<TeeKey>,
+    attester: Option<Box<dyn Attester + Send + Sync>>,
+}
+
+impl KbcInterface for Kbc {
+    fn check(&self) -> Result<KbcCheckInfo> {
+        Err(anyhow!("Check API of this KBC is unimplemented."))
+    }
+
+    fn decrypt_payload(&mut self, _annotation: &str) -> Result<Vec<u8>> {
+        Err(anyhow!("Decrypt Payload API of this KBC is unimplemented."))
+    }
+}
+
+impl Kbc {
+    pub fn new(_kbs_uri: String) -> Kbc {
+        // Detect TEE type of the current platform.
+        let tee_type = detect_tee_type().ok().unwrap_or(TEE::Unknown);
+
+        // Create attester instance.
+        let attester = tee_type.to_attester().ok();
```

Comment:
Errors in `detect_tee_type` or `to_attester` are swallowed and not reported anywhere. Shouldn't we completely fail in these cases? Why wait for `generate_evidence` to fail?

Or, thinking about this again: currently the Kbc struct has 3 Option fields. Can the Kbc work correctly when they are set to None? If not, they should not be Option; this way we won't construct a Kbc struct that will fail on every usage.

Reply:
Now according to the [KBC framework](https://github.com/confidential-containers/attestation-agent/blob/main/docs/kbc_module_development_guide.md), `fn new(_kbs_uri: String) -> Kbc` can not return a Result, so I have to temporarily use option in the KBC structure to surround the err returned by `Teekey::new()` and `to_attester()`. Maybe we can change the return value of the KBC's `new()` function later, but this will cause all KBCs in AA need to be modified.

Reply:
Yes, you're right. Thanks for pointing that out.

I also see that the KBC framework instantiates the kbc modules lazily, upon first request (for example in [`decrypt_image_layer_annotation()`](https://github.com/confidential-containers/attestation-agent/blob/main/src/lib.rs#L113)). Even if we change the return type of `new()` to `Result<Kbc>`, we'll encounter the error quite encounter the error quite late (and not an early fail when the attestation-agent starts, which is what I was trying to achieve).

I hope we can improve it later (in the framework level); but not in this PR.